### ISSUE TITLE
No progress updates when executing a batch upload with small files

### ISSUE THREAD

**Issue by gertjanvg** [dropbox/SwiftyDropbox#219]
I have been working on an application that uploads a bunch of photos to Dropbox. All of the photos are pretty small and I used the batch upload method available in Dropbox to upload all of them. Everything worked, up until the point I wanted to track the progress of the upload process. I passed a `BatchUploadResponseBlock` as parameter for `FilesRoutes.batchUploadFiles`, but no progress is being reported.

Diving into the source code revealed the following issue: there are two different types of file uploading: small files and large files. Since all of my files were small all of them were passed to the `startUploadSmallFile` method. It turned out that `startUploadSmallFile` does not actually ever call `self.executeProgressHandler`, which results in no progress being reported at all. 

I have not yet tested whether this lack of reporting results in incorrect progress reporting when mixing small and large files, but my guess is it does. This obviously is an issue because one would expect progress from an upload task, especially a batched task. 

For now I have put in place a quick bugfix, but a more elegant solution should be found. I will most likely submit a pull request later on with a better solution.

**Comment by greg-db** [comment]
Thanks for the report! We'll look into it.

**Comment by dor4emon** [comment]
Hi @gertjanvg , may I ask which batch upload method you are using? I can't find it
Thanks

**Comment by greg-db** [comment]
@dor4emon They are referring to [the `batchUploadFiles` method](https://github.com/dropbox/SwiftyDropbox/blob/345f460558cd1b1c4faf8c6c5fff80f5809044d1/Source/SwiftyDropbox/Shared/Handwritten/CustomRoutes.swift#L13). There's [an example in the test class](https://github.com/dropbox/SwiftyDropbox/blob/c51fea4e3afc1e4d7813d158fef661c130c6f7b2/TestSwiftyDropbox/IntegrationTests/TestClasses.swift#L56). Apologies this isn't properly documented! I'll ask the team to add some documentation.

**Comment by angelFrog** [comment]
This issue still exists.

As @gertjanvg says, the fix is quick and easy, just adding executeProgressHandler etc. to the end of startUploadSmallFile  to match the end of startUploadLargeFile.

**Comment by greg-db** [comment]
@angelFrog Thanks for the note! I'll bring this up with the team.

**Comment by pumPkin555** [comment]
I have tested as @angelFrog said and @gertjanvg guessed, but nothing happens. I still do not have progress. Can somebody help how to get progress?

**Comment by greg-db** [comment]
@pumPkin555 This issue is still open with the team, but it hasn't been addressed yet. I'll follow up here if/when I have an update on it.

**Comment by pumPkin555** [comment]
Thank you @greg-db! I look forward to your update.

**Comment by strong84** [comment]
Does it support batchUploadData instead of files?

**Comment by greg-db** [comment]
@strong84 No, the SDK doesn't offer a version of this method that takes `Data` objects instead of files, but I'll pass this along as a feature request. I can't promise if or when that might be implemented though.
